From cfc59113d46149bab027d6e7f2492540fd8822aa Mon Sep 17 00:00:00 2001
From: Uday Kumar Sundar <usundar@codeaurora.org>
Date: Mon, 7 Oct 2013 10:31:48 -0700
Subject: [PATCH 07/10] setting RLIMIT to INFINITY.

This is to generate coredumps for processes spawned by zygote

Change-Id: I2dbac76fbcaa7be3845ba5e12ad064548a25e2e3
---
 vm/native/dalvik_system_Zygote.cpp |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/vm/native/dalvik_system_Zygote.cpp b/vm/native/dalvik_system_Zygote.cpp
index 3dae6de..2834516 100644
--- a/vm/native/dalvik_system_Zygote.cpp
+++ b/vm/native/dalvik_system_Zygote.cpp
@@ -39,6 +39,10 @@
 #include <sys/utsname.h>
 #include <sys/capability.h>
 
+#ifdef HAVE_ANDROID_OS
+#include <cutils/properties.h>
+#endif
+
 #if defined(HAVE_PRCTL)
 # include <sys/prctl.h>
 #endif
@@ -407,8 +411,15 @@ static void enableDebugFeatures(u4 debugFlags)
             ALOGE("could not set dumpable bit flag for pid %d: %s",
                  getpid(), strerror(errno));
         } else {
+            char prop_value[PROPERTY_VALUE_MAX];
+            property_get("persist.debug.trace",prop_value,"0");
             struct rlimit rl;
-            rl.rlim_cur = 0;
+            if(prop_value[0] == '1') {
+                ALOGE("setting RLIM to infinity for process %d",getpid());
+                rl.rlim_cur = RLIM_INFINITY;
+            } else {
+                rl.rlim_cur = 0;
+            }
             rl.rlim_max = RLIM_INFINITY;
             if (setrlimit(RLIMIT_CORE, &rl) < 0) {
                 ALOGE("could not disable core file generation for pid %d: %s",
-- 
1.7.9.5

